src/service/node/balancing/load.cpp

```diff
@@ -168,3 +185,47 @@ load_balancer_t::balance() {
         overseer->spawn(pool);
     }
 }
+
+void
+load_balancer_t::on_idle_timer(const std::error_code& ec) {
+    COCAINE_LOG_TRACE(overseer->logger(), "idle timer triggered");
+
+    if (ec == asio::error::operation_aborted) {
+        COCAINE_LOG_TRACE(overseer->logger(), "idle timer, status: aborted");
+        return;
+    }
+
+    COCAINE_LOG_TRACE(overseer->logger(), "checking for idle slaves");
+
+    // TODO: Again, not ideal way to iterate over string->any map. Consider using types.
+    const auto info = overseer->info();
+    const auto pool = info["pool"].as_object();
+    const auto slaves = pool["slaves"].as_object();
+
+    for (const auto& kv : slaves) {
+        const auto& id = kv.first;
+        const auto& slave = kv.second.as_object();
+
+        const auto& state = slave["state"].as_string();
+        COCAINE_LOG_TRACE(overseer->logger(), "state: %s", state);
+
+        if (state == "active") {
+            const auto& load = slave["load"].as_uint();
+            const auto& last_channel_activity = slave["last_channel_activity"].as_int();
+
+            const auto profile = overseer->profile();
+
+            if (load == 0 && last_channel_activity > profile.timeout.idle) {
+                COCAINE_LOG_DEBUG(overseer->logger(), "found idle slave, despawning")(
+                    "id", id,
+                    "timeout", profile.timeout.idle
+                );
+
+                overseer->despawn(id, overseer_t::despawn_policy_t::graceful);
+            }
+        }
+    }
+
+    idle_timer.expires_from_now(boost::posix_time::seconds(1));
```

Comment:
I want to come up with idea to eval this period from slaves information. For example, idle_timeout - slave_with_max_idle_value could be used. But we should discuss it.